### PR TITLE
Dom crawler form examples

### PR DESCRIPTION
**Pull request by uikolas**
Added more examples, how to get crawler form.
I think for new users it's kinda useful, these examples. 
Especially if form button doens't contains name

### CODE REVIEW

File: components/dom_crawler.rst [hunk 1]
```diff
@@ -396,10 +396,19 @@ given text. This method is especially useful because you can use it to return
 a :class:`Symfony\\Component\\DomCrawler\\Form` object that represents the
 form that the button lives in::
 
-    $form = $crawler->selectButton('validate')->form();
+    // button example: <button id="my-super-button" type="submit">My super button</button>
+    
+    // you can get button my it's name
```
Comment (weaverryan) [inline components/dom_crawler.rst]: So, it matches by button text? What if it's an input type="submit" - does it use the name attribute then?

**Review by weaverryan — COMMENTED** [review]
A few comments, but I do like having the added examples!

**Comment by uikolas** [comment]
So, sorry for misspelling.

File: components/dom_crawler.rst (hunk 1 above)
Comment (uikolas) [inline components/dom_crawler.rst]: Sorry, I didn't understand what do you mean with: 

> What if it's an input type="submit" - does it use the name attribute then?

File: components/dom_crawler.rst [hunk 2]
```diff
@@ -396,10 +396,19 @@ given text. This method is especially useful because you can use it to return
 a :class:`Symfony\\Component\\DomCrawler\\Form` object that represents the
 form that the button lives in::
 
-    $form = $crawler->selectButton('validate')->form();
+    // button example: <button id="my-super-button" type="submit">My super button</button>
+    
+    // you can get button my its name
```
Comment (xabbuh) [inline components/dom_crawler.rst]: Actually, we are talking about its label, aren't we?

**Review by xabbuh — APPROVED** [review]

File: components/dom_crawler.rst [hunk 3]
```diff
@@ -396,10 +396,19 @@ given text. This method is especially useful because you can use it to return
 a :class:`Symfony\\Component\\DomCrawler\\Form` object that represents the
 form that the button lives in::
 
-    $form = $crawler->selectButton('validate')->form();
+    // button example: <button id="my-super-button" type="submit">My super button</button>
+    
+    // you can get button my its label
```
Comment (xabbuh) [inline components/dom_crawler.rst]: by

File: components/dom_crawler.rst [hunk 4]
```diff
@@ -396,10 +396,19 @@ given text. This method is especially useful because you can use it to return
...
+    $form = $crawler->selectButton('My super button')->form();
+    
+    // or by button id (#my-super-button) if button doesn't have a label
```
Comment (xabbuh) [inline components/dom_crawler.rst]: [...] if the button [...]

File: components/dom_crawler.rst [hunk 5]
```diff
@@ -396,10 +396,19 @@ given text. This method is especially useful because you can use it to return
...
+    $form = $crawler->selectButton('my-super-button')->form();
+    
+    // or you can filter whole form, for example a form has a class attribute: <form class="form-vertical" method="POST">
```
Comment (xabbuh) [inline components/dom_crawler.rst]: [...] filter the whole [...]

**Review by javiereguiluz — APPROVED** [review]

**Comment by weaverryan** [comment]
This is GREAT! I really like all the examples with comments to explain them. Easy to read and super clear. Thanks @uikolas!

**Pull request closed without merging**